### PR TITLE
ENG-7529/be config

### PR DESCRIPTION
**Pull request by cometman**

### CODE REVIEW

**Review by kishisaka-nid — APPROVED** [review]
LGTM. Tested. Saw config being pulled from proxy on startup (only once).

File: NeuroID/NeuroIDClass/Extensions/NIDSend.swift [hunk 1]
```diff
@@ -55,15 +55,20 @@ internal extension NeuroID {
...
+            if let workItem = NeuroID.sendGyroAccelCollectionWorkItem {
+                // Send up the first payload, and then setup a repeating timer
+                DispatchQueue
+                    .global(qos: .utility)
+                    .asyncAfter(
+                        deadline: .now() + Double(NIDConfigService.nidConfigCache.gyroAccelCadenceTime),
+                        execute: workItem
+                    )
+            }
+
```
Comment (quacke-nid) [inline NeuroID/NeuroIDClass/Extensions/NIDSend.swift:71]: Can you restore the indent here?

File: NeuroID/NeuroIDClass/NeuroID.swift [hunk 2]
```diff
@@ -103,18 +103,32 @@ public enum NeuroID {
...
+
+        
         networkMonitor = NetworkMonitoringService()
         networkMonitor?.startMonitoring()
 
         // begin gyro/accel sample rate
-        sendGyroAccelCollectionWorkItem = createGyroAccelCollectionWorkItem()
+        if (NIDConfigService.nidConfigCache.gyroAccelCadence) {
+            sendGyroAccelCollectionWorkItem = createGyroAccelCollectionWorkItem()
+        }
```
Comment (kevinsites) [inline NeuroID/NeuroIDClass/NeuroID.swift:131]: Will this be a race condition and should be included in LN 112-121?

File: NeuroID/Utilities/NIDConfigService.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+internal class NIDConfigService {
+    
+    struct ResponseData: Decodable {
+        var callInProgress: Bool = true
+        var geoLocation: Bool = true
+        var eventQueueFlushInterval: Int = 5
+        var eventQueueFlushSize: Int = 2000
+        var requestTimeout: Int = 10
+        var gyroAccelCadence: Bool = true
```
Comment (quacke-nid) [inline NeuroID/Utilities/NIDConfigService.swift:16]: Is the v1 implementation list at the end of the design doc the correct default values?  `gyro_accel_cadence` is marked as `false`.  Can you confirm the correct default?  @kishisaka-nid what is android default?

File: NeuroID/Utilities/NIDConfigService.swift [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+        enum CodingKeys: String, CodingKey {
+            case callInProgress = "call_in_progress"
+            case eventQueueFlushInterval = "event_queue_flush_interval"
+            case eventQueueFlushSize = "event_queue_flush_size"
+            case geoLocation = "geo_location"
+            case gyroAccelCadence = "gyro_accel_cadence"
+            case gyroAccelCadenceTime = "gyro_accel_cadence_time"
+            case requestTimeout = "request_timeout"
+        }
```
Comment (quacke-nid) [inline NeuroID/Utilities/NIDConfigService.swift:27]: Can you confirm if there is a test case for these mappings?  I don't see a local JSON, are the requests to neuro-dev exercising this?

File: NeuroID/Utilities/NIDConfigService.swift (hunk 3 above)
Comment (cometman) [reply NeuroID/Utilities/NIDConfigService.swift:16]: I misread. Good catch

File: NeuroID/Utilities/NIDConfigService.swift (hunk 4 above)
Comment (cometman) [reply NeuroID/Utilities/NIDConfigService.swift:27]: Test in place that encodes with all these mappings. Would fail if mappings did not match

**Review by kevinsites — APPROVED** [review]

**Review by quacke-nid — APPROVED** [review]

**Pull request merged**